### ISSUE TITLE
Better configuration

### ISSUE THREAD

**Issue by hydroid7** [s2e-systems/dust-dds#195]
In general, it is the configuration of DustDDS does not scale and is hard to work with. I'm opening this issue to discuss possible alternatives. 

**Inconsistent documentation on interfaces:**
The docs mention that you can pass an array of interface names as configuration option: `[string, null]`.
However, the implementation only accepts one `String` value xor `null`.

**Env variables does not scale well**
Environment variables are only ok for small number of config options and get cumbersome to use with great number of values.

**Possible solutions:**
- RTI Connext uses XML
- Eclipse CycloneDDS also uses XML

I think a TOML based configuration file would fit well into the rustacean philosophy. Then the configuration file path could be provided like `DUST_DDS_CONFIGURATION=/etc/dds/config.toml`.
I'd be glad to hear the opinion of the maintainers! 🙂

**Comment by jrebelo** [comment]
Thank you for opening this issue. It is a very interesting question and something we have indeed been recently considering how to handle.

Originally we picked for a JSON format because it is something we are personally used to and it was easy to integrate with our code given the dependencies we already had. It was also possible to generate a schema for it and use that schema for easy creation of configurations in the IDE and also the documentation. But as you mention in your issue description, the way JSON schemas are described are not so "compatible" with Rust (e.g. [String, null] meaning an option between a string or nothing and not an array of those things).

After giving some consideration to this we came to the view that it should not be the task of the DustDDS library to "impose" on our users a specific format for this configuration. Our preferred solution is to make the configuration available programmatically on the DomainParticipantFactory interface and then each application will have the freedom to import the configuration as they best see fit and use it.

We are still working on the technical solution but I will soon link the PR on this issue. It would be nice to get also your feedback on it.
